docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,606 @@
+example, this would be the section for a plugin `npe2-tester` with
+`napari.yaml` as the manifest file:
+
+```cfg
+[options.entry_points]
+napari.manifest =
+    npe2-tester = npe2_tester:napari.yaml
+```
+
+The manifest file is specified relative to the submodule root path.
+So for the example it will be loaded from: `<path/to/npe2-tester>/napari.yaml`.
+
+### 2. Include the manifest in the package distribution
+
+The manifest file needs to be included as _[package data][pd]_ in
+distributable forms for the package. For example:
+
+```toml
+[metadata]
+...
+include_package_data=True
+
+[options.package_data]
+npe2_tester =
+    napari.yaml
+```
+
+## Manifest schema
+
+The plugin manifest file is a [json][], [yaml][], or [toml][] conforming to
+a schema. The `npe2` package includes a command-line tool that can be used
+to validate a schema:
+
+```
+pip install npe2
+npe2 validate my_plugin_manifest.yml
+```
+
+The manifest file's fields are described in detail below. The manifest file's
+overall structure can be viewed as a hierarchy:
+
+```
+<top-level properties>
+contributions:
+  commands:     <list of commands>
+  readers:      <list of readers>
+  writers:      <list of writers>
+  sample_data:  <list of sample data providers>
+  widgets:      <list of widget providers>
+  themes:       <list of themes>
+```
+
+Readers, writers, sample data providers and widget providers refer to callable
+python functions that a plugin defines. Each callable is identified with an
+entry in the list of `commands` via a unique id. For more see {ref}`Commands`
+below.
+
+```{note}
+Python package metadata (`setup.py` or `setup.cfg`) may be used to populate
+some internal properties of the plugin manifest. These include the authors,
+description, and version.
+```
+
+(top-level-props)=
+
+## Top-level properties
+
+### Required
+
+- **name** The name of the plugin. Example: `napari-svg`. Should be a
+  [PEP-8]-compatible package name. If missing, this is populated from the
+  python package [name][setup-name].
+
+### Optional
+
+- **engine**: A SemVer compatible version string matching the versions of the
+  plugin engine that the extension is compatible with. Example: "0.1.0".
+  See [][compatibility] for details.
+- **display_name**: User-facing text to display as the name of this plugin.
+  Example: `napari SVG`. Must be 3-40 characters long, containing
+  printable word characters, and must not begin or end with an underscore,
+  white space, or non-word character.
+- **entry_point**: The module containing the `activate()` function. Example:
+  `foo.bar.baz`. This should be a fully qualified module string.
+
+```{admonition} activation and deactivation
+An `activate()` function can be used to perform one-time initialization for a
+plugin.  Similarly, a `deactivate()` function can be used to perform tear-down
+on deactivation.  These must be defined relative to the `entry_point`.
+```
+
+(compatibility)=
+
+### Compatibility
```

Comment:
there's a _lot_ of background here, which is perhaps of interest to napari developers looking to modify the schema, but a _lot_ for someone just writing a plugin who more or less just needs to know which number to use.

Reply:
I agree this is much too long. Can you suggest how you'd like to shrink it down? 

I kind of like keeping just the first paragraph. Probably with a different section title.